### DISCUSSION TITLE
Format in _metadata.yml not Merging Correctly

### DISCUSSION THREAD

**Discussion by jkub6** [quarto-dev/quarto-cli#8872] [Q&A] [answered]
### Description

I have options specified for the revealjs format within a folder's `_metadata.yml`. I also have an `index.qmd` file in this folder (html file). When rendering, Quarto tries to render this `index.qmd` file as a revealjs file, causing errors (conflicting html files)

Am I misunderstanding something, or is this a bug? From my reading, the documentation seems to imply that the formats specified in `metadata.yml` should be overridden by the format specified in `index.qmd`, but this does not appear to be happening.

[Metadata Merging:](https://quarto.org/docs/projects/quarto-projects.html#metadata-merging)

> The one exception to metadata merging is format. If the document-level YAML defines a format, it must define the complete list of formats to be rendered.

### File Structure
```
.
 |-_quarto.yml
 |-files
 | |-_metadata.yml
 | |-index.qmd
 | |-deck.qmd
```

### _quarto.yml

````yaml
project:
  output-dir: _build
````

### files/_metadata.yml

````yaml
format:
  revealjs:
    theme: dark
````

### files/index.qmd

````quarto
---
title: "index"
format: html
listing:
  type: default
  contents:
    - "*/*.qmd"
---

index file here
````

### files/deck.qmd

````quarto
---
title: "deck"
format: revealjs
---

## Test Deck
````

### Running `quarto render` results in:

```
[1/2] files/index.qmd
[2/2] files/deck.qmd

ERROR: NotFound: No such file or directory (os error 2): rename '/home/jake/projects/test/files/index.html' -> '/home/jake/projects/test/_build/files/index.html'

Stack trace:
    at Object.renameSync (ext:deno_fs/30_fs.js:207:7)
    at renderProject (file:///home/jake/repos/quarto-cli/src/command/render/project.ts:458:14)
    at eventLoopTick (ext:core/01_core.js:183:11)
    at async Command.fn (file:///home/jake/repos/quarto-cli/src/command/render/cmd.ts:248:26)
    at async Command.execute (file:///home/jake/repos/quarto-cli/src/vendor/deno.land/x/cliffy@v0.25.4/command/command.ts:1790:7)
    at async quarto (file:///home/jake/repos/quarto-cli/src/quarto.ts:149:3)
    at async file:///home/jake/repos/quarto-cli/src/quarto.ts:163:5
    at async mainRunner (file:///home/jake/repos/quarto-cli/src/core/main.ts:35:5)
    at async file:///home/jake/repos/quarto-cli/src/quarto.ts:153:3
```

**Comment by cscheid** [comment]
What quarto version are you using?

**Reply by jkub6** [reply]
I am currently using the main branch. I also have a Github action that appears to be failing for the same reason using the most recent release.

**Reply by mcanouil** [reply]
@jkub6 Could you share a "working" Git repository? I am not sure to get your setup right.
<img width="535" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/5cc07768-a4ac-41de-9f5a-697ddb781bbc">

**Reply by jkub6** [reply]
Sure, thanks for your time! Running `quarto render` from the root of this repo gives me the error. (I am on Linux, so maybe the path needs to be tweaked if you are on Windows)

https://github.com/jkub6/quarto-tmp

![image](https://github.com/quarto-dev/quarto-cli/assets/30987328/fa6030d4-95a5-44a5-90f2-25c3e390c1c4)

**Comment by cderv** [comment, marked as answer]
The "not found" error is related to this one :
* https://github.com/quarto-dev/quarto-cli/issues/6809

This is indeed because `index.qmd` tried to be rendered to HTML **and** reveals. 

> Am I misunderstanding something, or is this a bug? From my reading, the documentation seems to imply that the formats specified in metadata.yml should be overridden by the format specified in index.qmd, but this does not appear to be happening.

Our documentation is not clear enough on that but `_metadata.yml` is merged with the input document yaml, and this is what is considered to override the `_quarto.yml` config. FWIW I needed to check in the code base to be sure.  
https://github.com/quarto-dev/quarto-cli/blob/248a3a4b1cd8985eb062cd53e53b23326c13eb2d/src/command/render/render-contexts.ts#L418-L437

The current logic is the following: 

* Some project format requires format to be in the project config `_quarto.yml` (like books) so in that case, it is prioritized. 
* Otherwise, if format are defined in the merge of in-document yaml config and shared `_metadata.yml` directory config, they are prioritized. 
* At last, if none of the above, the formats are considered defined in `_quarto.yml`

This is like this from the very early versions

I'll push a clarification to the doc about this.  (see https://github.com/quarto-dev/quarto-cli/issues/8968)
